mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb

```diff
@@ -2324,11 +2338,22 @@ func resourceContainerClusterCreate(d *schema.ResourceData, meta interface{}) er
 		// Node Configs have default values that are set in the expand function,
 		// but can only be set if node pools are unspecified.
 		cluster.NodeConfig = expandNodeConfig([]interface{}{})
+		<% unless version == 'ga' -%>
```

Comment:
Why wouldn't users be able to specify `node_config` or `node_pool.node_config` to set these values? Are there bugs/issues you can link for context?

Reply:
CHD is a feature only in TF beta now. We will enable GA support in another PR.

Reply:
Oh, that part made sense- I picked this line because the comment applied to both fields we're overriding. Rephrased- why do we need to override these values when neither `node_config` or `node_pool.node_config` is specified rather than have them define those fields & specify the values?

Reply:
Because if the org policy is enabled that requires to use CHD, cluster creation via TF will not succeed because of this default node pool is always created without enabling CHD. 

We need a way to override this logic.

Reply:
Wouldn't either of those options work to satisfy that constraint? `node_config` changes the settings for the default pool & if `node_pool.node_config` is set there should be no default pool @ creation time.

Reply:
Hmm i am not sure about the no default pool @ creation time, because my understanding is that TF always create a default node pool and if you set the flag remove_default_node_pool to true, they will delete it afterwards.

This existing logic is causing the problem we are trying to solve in this PR.

Reply:
> because my understanding is that TF always create a default node pool

This is true unless there's at least one `node_pool` entry! That overrides the default pool.

> TF will delete it afterwards(this is why remove_default_node_pool exists to begin with). That is also the bug report from our customers about this logic is not working for them.

This can be mitigated w/ `lifecycle.ignore_changes`:

```
resource "google_container_cluster" "my-gke-cluster" {
  name     = "my-gke-cluster"
  location = "us-central1"

  # We can't create a cluster with no node pool defined, but we want to only use
  # separately managed node pools. So we create the smallest possible default
  # node pool and immediately delete it.
  remove_default_node_pool = true
  initial_node_count       = 1

  node_config {
    enable_confidential_storage = true
    disk_type = "hyperdisk-balanced"
    machine_type = "n2-standard-2"
    boot_disk_kms_key = "{{key}}""
  }

  lifecycle {
    ignore_changes = ["node_config"]
  }

  # other settings...
}
```

This configuration would create the default pool with the specified settings and then delete it, ignoring the diff indicating it has been deleted due to the lifecycle rule.

Instead of introducing these overrides we could stick an example somewhere showing how to do this, and drop a note on the fields under `node_config` referencing it. Admittedly this pushes complexity to the end user a bit- since they'll need a slightly more complex configuration- but stops us from adding a fourth point with `NodeConfig` settings that will grow in scope over time. There are other fields like service_account that would use the precedent for similar treatment, for example.

Reply:
If with the lifecycle {
    ignore_changes = ["node_config"]
  } setting the default node pool will be created with confidential then that is what we are asking for.

Reply:
Someone tested this a month back and it was working. The person will retest it today/tonight to make sure.

Reply:
I'm gonna close the PR in the meantime, we'll reopen if needed.